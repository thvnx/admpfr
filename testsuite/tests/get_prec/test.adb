with Ada.Exceptions; use Ada.Exceptions;
with Ada.Text_IO;    use Ada.Text_IO;
with Admpfr;         use Admpfr;

procedure Test is
   M : Mpfloat;
   N : Mpfloat (1);
begin
   Set (M, "1");
   Put_Line (Get_Prec (M)'Image);

   Put_Line (Get_Prec (N)'Image);

   begin
      declare
         O : Mpfloat (Precision'Last);
         --  16#7FFF_FFFF_FFFF_FEFF# if Precision is 8 bytes (unix)
      begin
         Put_Line (Get_Prec (O)'Image);
      end;
   exception
      when F : Failure => Put_Line (Exception_Message (F));
   end;
end Test;
