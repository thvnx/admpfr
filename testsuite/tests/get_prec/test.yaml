driver: admpfr
control:
- [XFAIL, "env.build.os.name == 'windows'",
   "no exception raised if long if 4 bytes and there is more than 4 Gb of memory (typically on windows)"]
- [XFAIL, "env.build.os.name == 'darwin'", "doesn't catch program error on macos"]
