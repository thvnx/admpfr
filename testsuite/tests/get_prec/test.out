 53
 1
GNU MP: Cannot allocate memory (size=1152921504606846760)
error during Mpfloat (Prec := 9223372036854774015) initialization
